core/src/error.rs

```diff
@@ -229,3 +256,27 @@ impl From<hyper::Error> for Error {
 		Error::Transport(hyper_err.into())
 	}
 }
+
+/// Internal error for the frontend/backend channel.
+#[derive(Debug)]
+pub enum InternalError {
```

Comment:
I like having our own error instead of using some SendError actually, even if it adds some code.